### PR TITLE
Update `CHANGELOG.md` for `6.2.4` release

### PR DESCRIPTION
**Pull request by adityapatwardhan**
# PR Summary

Update change log for 6.2.4

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: CHANGELOG/6.2.md [hunk 1]
```diff
@@ -1,5 +1,20 @@
 # 6.2 Changelog
```
Comment (PoshChan) [inline CHANGELOG/6.2.md]: @adityapatwardhan, your last commit had 1 failures in `PowerShell-CI-windows`
Job Cmdlet Tests.Retrieving partial output from jobs.Receive-Job will retrieve partial output, including -Keep results
```powershell
Destination array was not long enough. Check the destination index, length, and the array's lower bounds. (Parameter 'destinationArray')
at <ScriptBlock>, D:\a\1\s\test\powershell\Modules\Microsoft.PowerShell.Core\Job.Tests.ps1: line 108
108:                         $results = Receive-Job -Keep $job

```

File: CHANGELOG/6.2.md [hunk 2]
```diff
@@ -1,5 +1,20 @@
...
+- Enable `Start-Process` to work on Windows 7 (#10417) (Thanks @iSazonov!)
+- Fix global tool issues around exit code, command line parameters, and paths with spaces (#10461)
+- Make `Add-Type` usable in applications that host PowerShell (#10587)
+
+### Build and Packaging Improvements
+
+- Update to use `TSAv2` (#9914)
+- Update the dotnet SDK install script URL in `build.psm1` (#10927)
+- Update dependencies needed by Azure PowerShell and patch for `Newtonsoft.Json` (Internal 10798)
+- Fix path for getting reference assemblies (Internal 10792)
```
Comment (iSazonov) [inline CHANGELOG/6.2.md]: What is the "Internal"? It seems we have real number of PR.

Comment (TravisEz13) [reply CHANGELOG/6.2.md]: We use AzureDevOps to manage the releases for security reasons (as some releases have security changes.) . This is the PR# in that system.

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
@PoshChan Please remind me in 1 hour

**Comment by PoshChan** [comment]
@TravisEz13, this is the reminder you requested 1 hour ago

**Comment by TravisEz13** [comment]
@PoshChan  Please remind me in 1 hour

**Comment by PoshChan** [comment]
@TravisEz13, this is the reminder you requested 1 hour ago

**Pull request merged**